### PR TITLE
Move additional/block/insecure registires to /etc/containers/...

### PR DESCRIPTION
**Pull request by michaelgugino**
Move additional/block/insecure registires to /etc/containers/registries.conf

This commit moves additional/block/insecure registries to
/etc/containers/registries.conf and comments existing lines in
/etc/sysconfig/docker.

Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1460930

### CODE REVIEW

**Comment by michaelgugino** [comment]
Template from:  https://github.com/projectatomic/registries/blob/master/registries.conf

**Comment by ashcrow** [comment]
@michaelgugino does this replace ``daemon.json`` usage for ``docker`` backends?

**Comment by michaelgugino** [comment]
@ashcrow I'm unsure if system containers are reading that file or not.  According to the bugzilla link, docker should be utilizing this new file.  I believe crio will also read from this new file.  We'll have to investigate whether or not the change needs to be made to container-daemon.json for system containers.  In any case, that could be another PR and this PR doesn't make any changes to that part.

**Comment by ashcrow** [comment]
``container-daemon.json`` is docker's ``daemon.json`` configuration file which was to replace the old sysconfig location used by openshift-ansible. If ``/etc/containers/registries.conf`` replaces the registry portion for docker (and container engine which is docker) I think it would be a good idea to simplify into using one file (or set of files) for the configuration for all docker backends. It doesn't have to happen all in one PR, I just want to make sure we don't accidentally have the old sysconfig, the expected replacement of *daemon.json, and a new possibility in /etc/containers.

**Comment by ashcrow** [comment]
[test]

**Comment by michaelgugino** [comment]
@ashcrow unfortunately, this patch only affects registries; it's not a system-wide docker config file.  We're not getting rid of /etc/sysconfig/docker, someone decided to move only portions of the config (ie, the registires) to /etc/containers/registries.conf

daemon.json is only used in the following play (from docker role):
```
- name: Use System Container Docker if Requested
  include: systemcontainer_docker.yml
  when:
  - l_use_system_container
  - not l_use_crio_only
```

We'll need more input or investigation as to what actually consumes daemon.json and /etc/containers/registries.conf

**Comment by ashcrow** [comment]
flake https://github.com/openshift/origin/issues/15977

**Comment by ashcrow** [comment]
[test]

**Comment by baude** [comment]
FYI , for containers-image -> https://github.com/containers/image/pull/322

**Comment by ashcrow** [comment]
flake https://github.com/openshift/origin/issues/15977

**Comment by ashcrow** [comment]
aos-ci-test

**Comment by ashcrow** [comment]
infra flake
```
Unable to find image 'openshift3/ose:v3.6' locally\n/usr/bin/docker-current: Error response from daemon: Invalid hostname \"r\".\nSee '/usr/bin/docker-current run --help'.", 
```

**Comment by ashcrow** [comment]
aos-ci-test

File: roles/docker/tasks/package_docker.yml [hunk 1]
```diff
@@ -51,22 +51,22 @@
...
   register: docker_check
 
-- name: Set registry params
+- name: Comment old registry params in /etc/sysconfig/docker
   lineinfile:
     dest: /etc/sysconfig/docker
     regexp: '^{{ item.reg_conf_var }}=.*$'
-    line: "{{ item.reg_conf_var }}='{{ item.reg_fact_val | oo_prepend_strings_in_list(item.reg_flag ~ ' ') | join(' ') }}'"
-  when: item.reg_fact_val != '' and docker_check.stat.isreg is defined and docker_check.stat.isreg
+    line: "#{{ item.reg_conf_var }}=''# Moved to {{ containers_registries_conf_path }}"
```
Comment (ashcrow) [inline roles/docker/tasks/package_docker.yml]: 👍

**Comment by ashcrow** [comment]
Please rebase and we can test again.

**Pull request merged**